### DISCUSSION TITLE
STM32H563 Nucleo USB-C: Nothing Happens

### DISCUSSION THREAD

**Discussion by Lotharyx** [hathach/tinyusb#2292] [Q&A]
Hi.  I'm trying to use TinyUSB in a project built around the STM32H563 chip.  I'm testing on a Nucleo board, because one happens to exist with exactly that chip on it.  The Nucleo board has a USB-C connector populated for the USB peripheral on the micro.  I'm using the STM Cube toolchain (CubeMX, CubeIDE).

With tips gleaned from [this issue](https://github.com/hathach/tinyusb/issues/2203), I have code that compiles and runs, based largely on the cdc_msc_freertos example.  I can set breakpoints in the USB tasks and see that they reach points of waiting endlessly on empty queues.

I have a breakpoint immediately after `osal_queue_receive` in `tud_task_ext` and it is never hit, telling me that nothing is ever put into that queue.

When I plug in a USB-C cable between my computer and the Nucleo, nothing at all happens except for the VBUS LED lighting up (which is a hardware feature of the Nucleo, not under uC control).  The computer doesn't react at all; no enumeration or anything.

I have a breakpoint immediately after `osal_queue_receive` in `tud_task_ext` and it is never hit, telling me that nothing is ever put into that queue.

I'm wondering if maybe some interrupts aren't getting enabled, or something else related to the fact that the Nucleo has a USB-C connection (my project uses Type-B, but I don't have an assembled board in hand yet).

At this stage I'm not entirely sure what the right questions are to ask, so any tips or guidance at all will be greatly appreciated!

**Comment by MannyATX** [comment]
You may be having the same problem I had. 
See the thread called "USBTMC/USB488 and WinUSB query lockup" I posted.
